Shared/Samples/Show device location using indoor positioning/ShowDeviceLocationUsingIndoorPositioningView.Model.swift

```diff
@@ -0,0 +1,234 @@
+        
+        /// Counts the number of satellites which being used for the GPS location.
+        private var satelliteCount: Int = -1
+        
+        /// The value of the horizontal accuracy of the location (in meters).
+        private var horizontalAccuracy: Double = -1.0
+        
+        ///  This value tracks whether the source is GPS or BLE.
+        private var source: String = ""
+        
+        /// A indoors location data source based on sensor data, including but not
+        /// limited to radio, GPS, motion sensors.
+        private var indoorsLocationDataSource: IndoorsLocationDataSource?
+        
+        /// The map's location display.
+        private(set) var locationDisplay = LocationDisplay(dataSource: SystemLocationDataSource())
+        
+        /// The location manager which handles the location data.
+        private let locationManager = CLLocationManager()
+        
+        /// The measurement formatter for sensor accuracy.
+        private let measurementFormatter: MeasurementFormatter = {
+            let formatter = MeasurementFormatter()
+            formatter.unitStyle = .short
+            formatter.unitOptions = .providedUnit
+            formatter.numberFormatter.maximumFractionDigits = 1
+            formatter.numberFormatter.roundingMode = .up
+            return formatter
+        }()
+        
+        /// This is the published value of the data that is displayed.
+        @Published private(set) var labelTextLeading: String = ""
+        
+        /// This is the published value of the data that is displayed.
+        @Published private(set) var labelTextTrailing: String = ""
+        
+        /// Represents loading state of indoors data, blocks interaction until loaded.
+        @Published var isLoading = false
+        
+        /// Kicks off the logic loading the data for the indoors map and indoors location.
+        func loadIndoorData() async throws {
+            isLoading = true 
+            try await map.load()
+            try await setIndoorDatasource()
+            try await requestLocationServicesAuthorizationIfNecessary()
+        }
+        
+        /// Stops the location data source.
+        func stopLocationDataSource() {
+            Task {
+                await locationDisplay.dataSource.stop()
+            }
+        }
+        
+        /// A function that attempts to load an indoor definition attached to the map
+        /// and returns a boolean value based whether it is loaded.
+        /// - Parameter map: The map that contains the IndoorDefinition.
+        /// - Returns: A boolean value for whether the IndoorDefinition is loaded.
+        private func indoorDefinitionIsLoaded(map: Map) async throws -> Bool {
+            guard map.indoorPositioningDefinition?.loadStatus != .loaded else { return true }
+            try await map.indoorPositioningDefinition?.load()
+            return map.indoorPositioningDefinition?.loadStatus == .loaded
+        }
```

Comment:
This method seems to be extraneous. The sample only loads the web map once, so the load status will likely be either `notLoaded` -> `loaded` or -> `failed`. Also, if you really need the retry workflow, there is the `retryLoad` method for Loadable resources for certain specific situations.

I suggest move the nullability check of `map.indoorPositioningDefinition` and `try await map.indoorPositioningDefinition.load()` method call to the spot where it is called.